source/Utility/Logger.h

```diff
@@ -23,25 +62,49 @@ struct Log
 | static_cast<unsigned char>(LogType::WARNING)					\
 | static_cast<unsigned char>(LogType::MESSAGE)					)			
 
-	template <typename T>
-	static void Print(T log, const LogType type = LogType::MESSAGE)
+	static inline void Print(const LogType aType, const char* aFormat, ...)
 	{
-		// Checks if the input element is printable for std::cout.
-		static_assert(
-			std::is_arithmetic<T>::value
-			|| std::is_base_of<std::string, T>::value
-			|| std::is_same<const char*, T>::value, "Printed Element needs to be either a number, a string, or a character.");
-
-		if (((static_cast<char>(LogType::PROBLEM) & static_cast<char>(type)) & static_cast<char>(DebugLoggingTypes)) || DebugLoggingTypes == 0) std::cout << "(X)";													  		 
-		if (((static_cast<char>(LogType::SUCCESS) & static_cast<char>(type)) & static_cast<char>(DebugLoggingTypes)) || DebugLoggingTypes == 0) std::cout << "(V)";							  						  		  
-		if (((static_cast<char>(LogType::WARNING) & static_cast<char>(type)) & static_cast<char>(DebugLoggingTypes)) || DebugLoggingTypes == 0) std::cout << "(!)";							  								    
- 		if (((static_cast<char>(LogType::MESSAGE) & static_cast<char>(type)) & static_cast<char>(DebugLoggingTypes)) || DebugLoggingTypes == 0) std::cout << "( )";
-
-		bool shouldPrint = static_cast<char>(type) & static_cast<char>(DebugLoggingTypes);
+		std::string prefix = "";
+		switch (aType)
+		{
+			case LogType::PROBLEM:
+				prefix = "(P)";
+				break;
+			case LogType::SUCCESS:
+				prefix = "(S)";
+				break;
+			case LogType::WARNING:
+				prefix = "(W)";
+				break;
+			case LogType::MESSAGE:
+				prefix = "(M)";
+				break;
+			default:
+				break;
+		}
+
+		bool shouldPrint = static_cast<char>(aType) & static_cast<char>(DebugLoggingTypes);
 		if (shouldPrint || DebugLoggingTypes == 0)
 		{
-			std::cout << " " << log << "\n";
+			va_list argumentLeft, argumentRight;
+			va_start(argumentLeft, aFormat);
+			va_copy(argumentRight, argumentLeft);
+			int size = _vscprintf(aFormat, argumentRight) + 1;
+			va_end(argumentRight);
+
+			char* buffer = new char[size];
+			vsnprintf(buffer, size, aFormat, argumentLeft);
+			va_end(argumentLeft);
+
+			std::string time = GetCurrentDateTime(DateTimeFormat::Time);
+			std::string log(prefix + " " + time + " " + buffer);
+			Console::GetInstance().AddLog(log.c_str());
+
+			std::string filePath = GetCurrentDateTime(DateTimeFormat::Date) + ".txt";
+			std::ofstream outstream(filePath.c_str(), std::ios_base::out | std::ios_base::app);
+			outstream << time << '\t' << buffer << '\n';
+			outstream.close();
```

Comment:
We should keep in mind that this is very thread-unsafe, so if we want to be able to multithread this we'll have to make sure to add some mutexes, or maybe put this at the end of the frame.